`unneeded_throws_rethrows` false positive when `try` is in parameter in a function with trailing closure

### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

A clear and concise description of what the bug is. Ideally, provide a small (but compilable) example code snippet that
can be used to reproduce the issue.

```swift
import Foundation

func a() throws -> String {
    throw NSError(domain: "a", code: 1, userInfo: nil)
}

func b(text: String, completion: () -> Void) {
    print(text)
    completion()
}

func c() throws { // Violation on the throw here
    b(text: try a()) {
        print("")
    }
}
```

However, it is required by the compiler. Without the throw:

```bash
$ swift a.swift                                                           
a.swift:13:13: error: errors thrown from here are not handled
11 | 
12 | func c() {
13 |     b(text: try a()) {
   |             `- error: errors thrown from here are not handled
14 |         print("")
15 |     }
```

Without the closure, no violation is reported:
```swift
import Foundation

func a() throws -> String {
    throw NSError(domain: "a", code: 1, userInfo: nil)
}

func b(text: String) {
    print(text)
}

func c() throws { // All good
    b(text: try a())
}

```


```bash
$ swiftlint lint --only-rule unneeded_throws_rethrows --no-cache ./a.swift
Linting Swift files at paths ./a.swift
Linting 'a.swift' (1/1)
/Users/steffen/Desktop/a.swift:12:10: warning: Unneeded (Re)Throws Keyword Violation: Superfluous 'throws'; body of this function does not throw any error (unneeded_throws_rethrows)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure) 0.63.2
* Xcode version (run `xcodebuild -version` to be sure) Xcode 26.2 Build version 17C52
* Installation method used (Homebrew, CocoaPods, building from source, etc) Homebrew
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,6 +20,12 @@
   [SimplyDanny](https://github.com/SimplyDanny)
   [#6465](https://github.com/realm/SwiftLint/issues/6465)
 
+* Take `try` expressions in call parameters into account even if the call has
+  trailing closures as well, so that an outer `throws` is not considered unneeded
+  in the `unneeded_throws_rethrows` rule.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#6491](https://github.com/realm/SwiftLint/issues/6491)
+
 * Ensure that disable commands work for `prefer_key_path` rule when the trailing
   closure starts on a different line than the function call as a whole.  
   [SimplyDanny](https://github.com/SimplyDanny)

--- a/Source/SwiftLintBuiltInRules/Rules/Lint/UnneededThrowsRule.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Lint/UnneededThrowsRule.swift
@@ -112,14 +112,20 @@ private extension UnneededThrowsRule {
         }
 
         override func visit(_ node: FunctionCallExprSyntax) -> SyntaxVisitorContinueKind {
-            if node.containsClosureDeclaration {
+            walk(node.calledExpression)
+            walk(node.arguments)
+            if node.hasTrailingClosures {
                 scopes.openScope()
+                if let trailingClosure = node.trailingClosure {
+                    walk(trailingClosure)
+                }
+                walk(node.additionalTrailingClosures)
             }
-            return .visitChildren
+            return .skipChildren
         }
 
         override func visitPost(_ node: FunctionCallExprSyntax) {
-            if node.containsClosureDeclaration {
+            if node.hasTrailingClosures {
                 scopes.closeScope()
             }
         }
@@ -195,8 +201,8 @@ private extension Stack where Element == UnneededThrowsRule.Scope {
 }
 
 private extension FunctionCallExprSyntax {
-    var containsClosureDeclaration: Bool {
-        children(viewMode: .sourceAccurate).contains { $0.is(ClosureExprSyntax.self) }
+    var hasTrailingClosures: Bool {
+        trailingClosure != nil || additionalTrailingClosures.isNotEmpty
     }
 }
 

--- a/Source/SwiftLintBuiltInRules/Rules/Lint/UnneededThrowsRuleExamples.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Lint/UnneededThrowsRuleExamples.swift
@@ -1,4 +1,6 @@
-internal struct UnneededThrowsRuleExamples {
+// swiftlint:disable file_length
+
+struct UnneededThrowsRuleExamples { // swiftlint:disable:this type_body_length
     static let nonTriggeringExamples = [
         Example("""
             func foo() throws {
@@ -154,6 +156,21 @@ internal struct UnneededThrowsRuleExamples {
         Example("let foo: () throws -> Void = bar"),
         Example("var foo: () throws -> Void = {}"),
         Example("let x = { () throws -> Void in try baz() }"),
+        Example("""
+        func c() throws {
+            b(text: try a()) { print("") }
+        }
+        """),
+        Example("""
+        func c() throws {
+            b(text: try a())
+        }
+        """),
+        Example("""
+        func c() throws {
+            [try f()].map { $0 }
+        }
+        """),
     ]
 
     static let triggeringExamples = [
@@ -313,6 +330,12 @@ internal struct UnneededThrowsRuleExamples {
         }
         """),
         Example("let x = { () ↓throws -> Void in baz() }"),
+        Example("""
+        func c() {
+            b { (n: String) ↓throws -> String in n }
+            d: { try foo() }
+        }
+        """),
     ]
 
     static let corrections = [

